### PR TITLE
Fix Android text truncation problem

### PR DESCRIPTION
**Pull request by LinYunMo**
Re: https://github.com/cocos/3d-tasks/issues/16856

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

File: cocos/2d/assembler/label/text-processing.ts [hunk 1]
```diff
@@ -88,6 +88,8 @@ export class TextProcessing {
     public processingString (isBmFont: boolean, style: TextStyle, layout: TextLayout,
         outputLayoutData: TextOutputLayoutData, inputString: string, out?: string[]) {
         if (!isBmFont) {
+            this._fontScale = this._getStyleFontScale(style.fontSize, style.fontScale,
+                outputLayoutData.canvasSize.width, outputLayoutData.canvasSize.height);
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-processing.ts]: outputLayoutData.canvasSize has any data ? can not use it before setting it.

File: cocos/2d/assembler/label/text-processing.ts [hunk 2]
```diff
@@ -141,12 +143,13 @@ export class TextProcessing {
 
     private _getStyleFontScale (fontSize: number, fontScale: number, canvasWidth: number, canvasHeight: number) {
         let scale = fontScale;
+        scale = parseFloat(scale.toFixed(1));
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-processing.ts]: Use `Math.round`.

File: cocos/2d/assembler/label/text-processing.ts [hunk 3]
```diff
@@ -159,7 +162,7 @@ export class TextProcessing {
             return;
         }
 
-        style.actualFontSize = style.fontSize;
+        style.actualFontSize = style.fontSize * this._fontScale;
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-processing.ts:157]: Seems like actualFontSize is used internally. no needs to set it to style.

File: cocos/2d/assembler/label/text-processing.ts [hunk 4]
```diff
@@ -198,15 +202,29 @@ export class TextProcessing {
...
             // set node height
-            outputLayoutData.nodeContentSize.height = rawHeight + outputLayoutData.contentSizeExtend.height;
+            outputLayoutData.nodeContentSize.height = (rawHeight + outputLayoutData.contentSizeExtend.height * this._fontScale) / this._fontScale;
             break;
         }
         default: {
             // nop
         }
         }
+        style.fontDesc = style.fontDesc.replace(
```
Comment (SantyWang) [inline cocos/2d/assembler/label/text-processing.ts]: fontDesc is used internally. no need to set back to style.

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Pull request merged**